### DISCUSSION TITLE
Alert Message - Telegram

### DISCUSSION THREAD

**Discussion by Lockie85** [openobserve/openobserve#3836] [Q&A]
Hey,

I've setup an Alert which when triggered sends a message over Telegram.

Here's my Template:
```
{
  "text": "Alert: Log entry matching criteria found!\nStream: {stream_name}\nAlert: {alert_name}\nDetails: {rows}",
  "rows_template": "{timestamp}: {message}"
}
```

Here's my Destination:
`https://api.telegram.org/bot<YourBotToken>/sendMessage?chat_id=<YourChatID>&text={{text}}`

The result I was hoping for was my Telegram message includes the "text" entry in my Template. However, what I get in the alert is the literal string {{text}}

Can anyone help me understand what I should put in order to see the template text details in the Telegram message?

Many Thanks

**Comment by prabhatsharma** [comment]
Not familiar with but can you try POST instead of GET.

**Reply by Lockie85** [reply]
The destination is already set to POST.

**Reply by prabhatsharma** [reply]
Then the text will be sent as request body and not via URL querystring. You have placed the text as a query parameter

**Reply by Lockie85** [reply]
Not sure I follow sorry. I'm a bit of a n00b.

Can you provide an example of how the URL needs to look in order to get my Template into my Telegram message?

**Reply by prabhatsharma** [reply]
Something like this - 

```
curl -X POST "https://api.telegram.org/bot<YourBotToken>/sendMessage" \
     -H "Content-Type: application/json" \
     -d '{"chat_id": "<ChatID>", "text": "Hello, this is a test message"}'
```

**Reply by Lockie85** [reply]
Thanks for the help so far but I don't think that is what I need as I don't think I've been clear on my issue.

I've got a ingestion of syslog and within those logs each entry contains a "message" field. Here's an example entry:
`{"_timestamp":1719513853000000,"appname":"unbound","facility":"daemon","message":"reply: 192.168.1.19 pool.ntp.org. A IN NOERROR 0.003129 0 94","procid":53411,"severity":"info"}`

I've then got an alert which reads as follows:
```
SELECT *
FROM DNSResolver
WHERE message LIKE '%car%'
```
This will trigger whenever it finds the string "car" within a log entry within the "message" field.

What I want to achieve with the "Destination" > "URL" which triggers the Telegram message that you note above. Is to insert the entire "message" field and its content from the entry triggering the alert..

**Reply by prabhatsharma** [reply]
While I cannot tell you specifics of telegram (I have never used it), but you can achieve the desired result for any destination. Check this video for specifics on how to do it - https://www.youtube.com/watch?v=tW8VnNnfZBg
